hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/PrefetchExecutor.java

```diff
@@ -79,6 +88,12 @@ public Thread newThread(Runnable r) {
       + HConstants.HREGION_COMPACTIONDIR_NAME.replace(".", "\\.") + Path.SEPARATOR_CHAR + ")");
 
   public static void request(Path path, Runnable runnable) {
+    if(!prefetchCompleted.isEmpty()){
```

Comment:
nit: No need for this extra check, as `prefetchCompleted.containsKey` would return false anyways if the map is empty.